packages/create-next-app/helpers/examples.ts

```diff
@@ -2,20 +2,67 @@ import got from 'got'
 import promisePipe from 'promisepipe'
 import tar from 'tar'
 
-export async function hasExample(name: string): Promise<boolean> {
-  const res = await got(
+export type RepoInfo = {
+  username: string
+  name: string
+  branch: string
+  filePath: string
+}
+
+export async function isUrlOk(url: string) {
+  const res = await got(url).catch(e => e)
+  return res.statusCode === 200
+}
+
+export function getRepoInfo(
+  url: URL,
+  examplePath?: string
+): RepoInfo | undefined {
+  const [, username, name, tree, _branch, ...path] = url.pathname.split('/')
+  const filePath = examplePath || path.join('/')
+  // If examplePath is available, the branch name takes the entire path
+  const branch = examplePath ? _branch + path.join('/') : _branch
```

Comment:
What if the user includes the pathname in the URL?

```
--example="https://github.com/zeit/next-learn-demo/foo/bar/baz" --example-path baz
```

In this case, `branch` needs to be `foo/bar` instead of `foo/bar/baz`.

Maybe:

```js
// If examplePath is available, the branch name takes the entire path
// excluding examplePath at the end
const pathRegex = new RegExp(`/${examplePath}$`)
const branch = examplePath ? (_branch + path.join('/')).replace(pathRegex, '') : _branch
```

Reply:
Why would the user add it twice?

Reply:
They might copy paste the entire URL (including the directory name) 🤔